sematic/plugins/storage/memory_storage.py

```diff
@@ -48,26 +49,57 @@ def get(cls, key: str):
     def set(cls, key: str, value: bytes):
         cls._store[key] = value
 
-    def get_write_location(self, namespace: str, key: str) -> str:
-        return f"/memory_upload/{namespace}/{key}"
+    def get_write_location(
+        self, namespace: str, key: str, user: Optional[User]
+    ) -> Location:
+        return Location(
+            location=f"http://localhost:5001/api/v1/uploads/{namespace}/{key}/memory",
+            headers=_make_headers(user),
+        )
 
-    def get_read_payload(self, namespace: str, key: str) -> ReadPayload:
-        try:
-            content = self._store[self.get_write_location(namespace, key)]
-        except KeyError:
-            raise NoSuchStorageKeyError(self.__class__, key)
+    def get_read_location(
+        self, namespace: str, key: str, user: Optional[User]
+    ) -> Location:
+        return Location(
+            location=f"http://localhost:5001/api/v1/uploads/{namespace}/{key}/memory",
```

Comment:
Please get the address and port from the config. Otherwise customizing the local config will result in execution failure for pipelines.